pcdsdevices/interface.py

```diff
@@ -217,9 +218,162 @@ def __init_subclass__(cls, **kwargs):
+            proceeding. Set this to False in cases where
+            you don't need the values immediately.
+        """
+        if not self._active_flag:
+            try:
+                self.ensure_cpts(wait_connected=wait_connected)
+            except TimeoutError:
+                pass
+
+    def ensure_cpts(
+        self,
+        cpts: Optional[Iterable[Union[Component, str]]] = None,
+        include_lazy: bool = False,
+        ensure_subs: bool = True,
+        wait_connected: bool = False,
+        timeout: Optional[float] = None,
+    ):
+        """
+        Make sure that the given components are ready to use.
+
+        Calling with no arguments is equivalent to the startup normally
+        done in the base Device.__init__ that is skipped for this interface.
+
+        Parameters
+        ----------
+        cpts : iterable of Components or str, optional
+            Either the component objects themselves or strings that match the
+            attrs of the Components. If omitted, we'll go through all of the
+            Components.
+        include_lazy : bool, optional
+            If True, we'll also ensure the lazy components are ready.
+            Defaults to False, which means ignore the lazy components.
+        ensure_subs : bool, optional
+            If True, we'll instantiate the components that have subs, even if
+            they are lazy.
+            Defaults to True.
+        wait_connected : bool, optional
+            If True, we'll wait for everything to connect and raise a
+            TimeoutError if anything does not connect during the timeout
+            duration.
+            Defaults to False.
+        timeout : float, optional
+            The maximum time to wait for connections if wait_connected
+            is True. Defaults to 0.005s per component, or at least 1s.
+        """
+        if cpts is None:
+            activation_call = True
+            try:
+                cpts: Iterable[Component] = self._sig_attrs.values()
+            except AttributeError:
+                # No _sig_attrs means not a Device, meaning no cpts
+                self._active_flag = True
+                return
+            if wait_connected and timeout is None:
+                timeout = max(1.0, 0.005 * len(list(self.walk_components())))
+        else:
+            activation_call = False
+        if wait_connected and timeout is None:
+            timeout = max(1.0, 0.005 * len(cpts))
+        start = time.monotonic()
+        objs: List[Union[Device, Signal]] = []
+        for cpt in cpts:
+            if isinstance(cpt, str):
+                cpt: Component = self._sig_attrs[cpt]
+            if any((
+                cpt._subscriptions and ensure_subs,
+                include_lazy,
+                not cpt.lazy,
+            )):
+                objs.append(getattr(self, cpt.attr))
+        if activation_call:
+            self._active_flag = True
```

Comment:
It's a bit messy exactly as you've described. The design goal here was to never repeat the "full device activate" sort of thing, but I didn't want to add overhead where I try to figure out which signals still need to be checked... maybe it could be added with none/minimal slowdown.